app/models/response_round.rb

```diff
@@ -2,4 +2,11 @@ class ResponseRound < ApplicationRecord
 	belongs_to :consultation, optional: true
 	has_many :questions
 	has_many :respondents
+	after_commit :add_response_round, on: :create
```

Comment:
@BALAJI-BALA - why not `before_commit` on: :create. Then you have only 1 SQL create query sent